packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -103,6 +103,8 @@ class _ModalBottomSheetLayout extends OneChildLayoutDelegate {
     childTop.end = childSize.height;
     return new Point(0.0, size.height - childTop.value);
   }
+
+  bool shouldRelayout() => true;
```

Comment:
This function should be passed the oldDelegate so that it can make a meaningful decision.